atoma-proxy/src/server/middleware.rs

```diff
@@ -345,36 +353,70 @@ pub async fn authenticate_middleware(
         })
         .await?;
 
-        STACK_NUM_REQUESTS_COUNTER.add(1, &[KeyValue::new("stack_small_id", stack_small_id)]);
+        if let Some(stack_small_id) = &stack_small_id {
+            STACK_NUM_REQUESTS_COUNTER.add(1, &[KeyValue::new("stack_small_id", *stack_small_id)]);
+        }
 
         // Validates the stack for the request.
         //
         // NOTE: If this method fails, we need to rollback the compute units that we locked for the stack, back to 0. Otherwise,
         // the proxy will be in an inconsistent state for the current stack.
-        let req = match utils::try_validate_stack_for_request(
-            &state,
-            &body_json,
-            &mut req_parts,
-            selected_node_id,
-            stack_small_id,
-            num_input_compute_units,
-            max_total_compute_units,
-            tx_digest,
-            user_id,
-            &endpoint,
-        )
-        .await
-        {
-            Ok(req) => req,
-            Err(e) => {
-                update_state_manager(
-                    &state.state_manager_sender,
+        let req = match stack_small_id {
+            Some(stack_small_id) => {
+                match utils::try_validate_stack_for_request(
+                    &state,
+                    &body_json,
+                    &mut req_parts,
+                    selected_node_id,
                     stack_small_id,
-                    max_total_compute_units as i64,
-                    0,
```

Comment:
What was the 0 for ? And why don't we keep it ?

Reply:
It's now couple lines before. This is full refund, We charge 0 tokens/usd in case of failure on our side.